tasks/getters/getActiveTaskPlans.js

```diff
@@ -1,13 +1,17 @@
 import { createSelector } from 'reselect'
-import { filter, isNil, values } from 'ramda'
+import { map, prop, filter, contains, not, values } from 'ramda'
 
-import getTaskPlans from './getTaskPlans'
+import getParentTaskPlans from './getParentTaskPlans'
+import getRawTaskWorks from './getRawTaskWorks'
 
 const getActiveTaskPlans = createSelector(
-  getTaskPlans,
-  (taskPlans) => {
-    const byDueDate = n => isNil(n.dueDate)
-    return filter(byDueDate, values(taskPlans))
+  getParentTaskPlans,
+  getRawTaskWorks,
+  (taskPlans, taskWorks) => {
+    const taskWorksTaskPlanIds = map(prop('taskPlanId'), values(taskWorks))
```

Comment:
rather than doing the check for "isActive" in a separate getter, i think we should put this logic in the getter that enhances task plans, so for each task plan we add a derived property `isActive`. then if we want this getter, it's as easy as filtering by `isActive`. the difference is that with this current approach, if we are using `getTaskPlans` then we don't know which ones are active.

Reply:
https://github.com/root-systems/cobuy/blob/master/tasks/getters/getEnhancedTaskPlans.js

Reply:
hey, rather than using `ramda` as if it were `lodash` with the arguments in a strange reverse order, i recommend using `ramda` in two stages:

1. create a function without the data (this should be done at the highest scope possible, as in if it doesn't rely on any data variables then it can be done in the top-level scope next to where you have imports and such)
2. use the function on the data

so in this case, this code can be written as:

```js
// top-level scope
const getTaskPlanIdsFromTaskWorks = map(prop('taskPlanId'))

// in function scope (with data)
const taskPlanIds = getTaskPlanIdsFromTaskWorks(taskWorks)
```